File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        // sort(nums.begin(),nums.end());
        // // int c=0,maxc=0,res=nums[0];
        // // for(int i=1;i<nums.size();i++){
        // //     if(nums[i]==nums[i-1])
        // //         c++;
        // //     else c=0;
        // //     maxc=max(maxc,c);
        // //     res=nums[maxc];
        // // }
        // // return res;
        
        
        // return nums[nums.size()/2];
        
//         without sorting the array:
        unordered_map<int,int> freq;
        for(int n:nums)
            freq[n]++;
        int n=nums.size()/2;
        for(auto x:freq)
            if(x.second>n)
                return x.first;
        return 0;
    }
};

File: 0169-majority-element/0169-majority-element.py
class Solution(object):
    def majorityElement(self, nums):
        # rec={}
        # for i in nums:
        #     rec[i]=rec.get(i,0)+1
        #     if rec[i]>len(nums)//2: return i

        # nums.sort()
        # return nums[len(nums)//2]
        no,c= None,0
        for n in nums:
            if no==n: c+=1
            elif c==0:
                no=n
                c=1
            else: c-=1
        return no
        """
        :type nums: List[int]
        :rtype: int
        """



        